src/client.ts

```diff
@@ -16,109 +34,189 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
-
-    for (const user of users) {
-      await iteratee(user);
+  public async getAccount(): Promise<Account> {
+    const uri = '/api/account/get-account';
+    const request = got.post(BASE_URI + uri, {
+      headers: this.generateHeaders(uri),
+      body: EMPTY_BODY,
+    });
+    try {
+      const result = await request;
+      const response = JSON.parse(result.body) as ApiResponse<Account>;
+      this.verifyResponse(response);
+      return response.data[0];
+    } catch (err) {
+      throw new IntegrationProviderAPIError({
+        cause: err,
+        endpoint: BASE_URI + uri,
+        status: err.status,
+        statusText: err.statusText,
+      });
     }
   }
 
-  /**
-   * Iterates each group resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
-   */
-  public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
+  public async getDomains(): Promise<Domain[]> {
+    const uri = '/api/domain/get-internal-domain';
+    const request = got.post(BASE_URI + uri, {
+      headers: this.generateHeaders(uri),
+      body: EMPTY_BODY,
+    });
+    try {
+      const result = await request;
+      const response = JSON.parse(result.body) as ApiResponse<Domain>;
+      this.verifyResponse(response);
+      return response.data;
+    } catch (err) {
+      throw new IntegrationProviderAPIError({
+        cause: err,
+        endpoint: BASE_URI + uri,
+        status: err.status,
+        statusText: err.statusText,
+      });
+    }
+  }
 
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
+  // TODO: pagination support
+  public async getUsers(domain: string): Promise<User[]> {
+    const uri = '/api/user/get-internal-users';
+    const request = got.post(BASE_URI + uri, {
+      headers: this.generateHeaders(uri),
+      body: JSON.stringify({
+        data: [
           {
-            id: 'acme-user-1',
+            domain,
           },
         ],
-      },
-    ];
+      }),
+    });
+    try {
+      const result = await request;
+      const response = JSON.parse(result.body) as ApiResponse<UserResponse>;
+      this.verifyResponse(response);
+      return response.data[0].users;
```

Comment:
So this and the getAccount calls are supposed to return just the first value in the data array?

Reply:
Yes. It is quite abnormal.

Reply:
even more so here, considering that users is the desired array of users. It should be structured like the `domain` response, in which the top level `data` property is the array of entities returned. But for this endpoint, there is this additional nesting for reasons unknown.

Reply:
So it's just literally an array of one item?  I wonder if it's set up that way for when/if our credentials are authorized for multiple accounts.  So you'd get a list of users for each account.  Anyway, just speculation on my part.  Mainly wanted to call it out just to make sure it was deliberate.  Thanks!